### PR TITLE
Create list and detail view for products

### PR DESCRIPTION
**Pull request by Karthikeyantestpress**

### CODE REVIEW

File: myshop/tests/shop/test_absolutepath.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from django.test import TestCase
```
Comment (waranlogesh) [inline myshop/tests/shop/test_absolutepath.py]: test_model_methods.py

File: myshop/tests/shop/test_absolutepath.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from django.test import TestCase
+from .test_modelmixintestcase import ModelMixinTestCase
+from django.urls import reverse
+
+
+class Test_model_method(ModelMixinTestCase, TestCase):
```
Comment (waranlogesh) [inline myshop/tests/shop/test_absolutepath.py]:
```suggestion
class TestModelMethod(ModelMixinTestCase, TestCase):

```

File: myshop/tests/shop/test_modelmixintestcase.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from django.test import TestCase, Client
```
Comment (waranlogesh) [inline myshop/tests/shop/test_modelmixintestcase.py]: Underscore separate each words in the file name

File: myshop/tests/shop/test_views.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        self.assertTemplateUsed(response, "shop/product/list.html")
+
+    def test_templates_used_for_product_detail_view(self):
+        self.product_list_url = reverse(
+            "shop:product_detail",
+            args=[self.newproduct.id, self.newproduct.slug],
+        )
+
+        response = self.client.get(self.product_list_url)
+        self.assertTemplateUsed(response, "shop/product/detail.html")
```
Comment (waranlogesh) [inline myshop/tests/shop/test_views.py:21]: test_should_return_404_for_invalid_slug

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: myshop/tests/shop/test_views.py (hunk 4 above)
Comment (Karthikeyantestpress) [reply myshop/tests/shop/test_views.py:21]: Updated Testcases  to return 404 for invalid slug

File: myshop/tests/shop/test_modelmixintestcase.py (hunk 3 above)
Comment (Karthikeyantestpress) [reply myshop/tests/shop/test_modelmixintestcase.py]: Updated

**Review by waranlogesh — APPROVED** [review]

**Pull request merged**